### PR TITLE
Fix nomatch index crash

### PR DESCRIPTION
**Pull request by erykoff**
I fixed a crash when there are no matches on query_radius and return_indices=True.  I also added some additional tests, and finally import Matcher at the module level.

### CODE REVIEW

**Comment by erykoff** [comment]
Oh!  And also, presumably due to some numpy changes, the tests as they were written were failing.  I increased the tolerances, but there are definitely some issues near 0 distance with the `sphdist` implementation having to do with `np.rad2deg(np.arccos(cosd))`.

**Comment by erykoff** [comment]
(I noticed the regular failing tests when I forked the repo and the workflow failed, so it's definitely not just my local computer which saw the same failures.)

File: smatch/tests/test_matcher.py [hunk 1]
```diff
@@ -106,6 +119,61 @@ def test_matcher_radius():
...
+    decp = dec[::-1] + 0.1
+
+    idx, i1, i2, d = mch.query_radius(rap, decp, 0.2, return_indices=True)
+
+    assert np.max(d) < 0.2
+    dist_match = sphdist(ra[i1], dec[i1], rap[i2], decp[i2])
+    assert np.max(dist_match) < 0.2
+
+
+def test_matcher_radius_indices():
```
Comment (beckermr) [inline smatch/tests/test_matcher.py]: Why is this test repeated?

**Review by esheldon — APPROVED** [review]
Thanks Eli.

**Review by beckermr — CHANGES_REQUESTED** [review]
one comment

File: smatch/tests/test_matcher.py (hunk 1 above)
Comment (erykoff) [reply smatch/tests/test_matcher.py]: Because it's such a damn important test!

Or because of a cut-and-paste error.

**Review by beckermr — APPROVED** [review]

**Pull request merged**